app/models/miq_server/worker_management/monitor.rb

```diff
@@ -93,25 +93,11 @@ def do_system_limit_exceeded
   end
 
   def sync_monitor
-    @last_sync ||= Time.now.utc
+    @last_sync          ||= Time.now.utc
     sync_interval         = @worker_monitor_settings[:sync_interval] || 30.minutes
     sync_interval_reached = sync_interval.seconds.ago.utc > @last_sync
-    roles_changed         = my_server.active_roles_changed?
-    resync_needed         = roles_changed || sync_interval_reached
```

Comment:
Once you remove `resync_needed`, then this splits up nicely and it is easy to pull `roles_changed` code out of here.